### DISCUSSION TITLE
Is it possible to customise Caddy settings?

### DISCUSSION THREAD

**Discussion by timsluis** [dokku/dokku#5540] [Q&A] [answered]
I would like to create a reverse proxy from subdirectory to an API, say /api/* to https://my.api.com. The API is another Dokku app on the same server.

I tried creating labels using docker-options, but I can't get it to work. Is there a way to customise the Caddy configuration on a per app basis?

**Comment by josegonzalez** [comment, marked as answer]
There is an example in the upstream issue tracker [here](https://github.com/lucaslorentz/caddy-docker-proxy/issues/208#issuecomment-761845919) that may be what you are looking for.

Note that the caddy configuration is generated via labels, and subject to the engine provided by caddy-docker-proxy.

**Reply by timsluis** [reply]
Thanks for the reply @josegonzalez! 

So just to be sure: these labels are set via docker-options in the deploy phase? Also, is there a way to see the resulting docker-compose file?

**Reply by josegonzalez** [reply]
Yeah, the labels are added that way - probably should be documented 😅 - to your app's running containers (the `deploy` phase is the correct one).

There is no "compose file" for your app containers, only for the caddy proxy. This can be displayed via `caddy:show-config`, as documented [here](https://dokku.com/docs/networking/proxies/caddy/#showing-the-caddy-compose-config).

**Reply by timsluis** [reply]
@josegonzalez When I try to add a label using for example this command: `dokku docker-options:add my-app deploy --label=caddy.lb_policy:cookie` it results in a `ERR_SSL_PROTOCOL_ERROR`. I tried all kinds of combinations but I always get this error. Do you know what I am doing wrong?

**Reply by josegonzalez** [reply]
Did you enable ssl for the app? Thats the only thing I can think of re: that ssl protocol error. Hard to say without looking at your config/checking the domain. Feel free to jump on our slack for more live support :)

**Reply by timsluis** [reply]
Will do, thanks!
